Assets/Scripts/Editor/Compilation/CodeRewriting/Partials/PartialTreeProcessor.cs

```diff
@@ -67,25 +86,41 @@ private static bool HasPartialTypes(SyntaxTree tree)
 
         private static void ProcessTree(
                 SyntaxTree tree,
-                Dictionary<string, List<TypeDeclarationSyntax>> combinedTypes,
-                HashSet<UsingDirectiveSyntax> combinedUsingDirectives,
-                HashSet<string> combinedTypesDefined)
+                Dictionary<string, Dictionary<string, List<TypeDeclarationSyntax>>> combinedTypes,
+                (HashSet<string> strings, HashSet<UsingDirectiveSyntax> syntaxes) combinedUsings)
         {
             var root = tree.GetCompilationUnitRoot();
 
-            combinedUsingDirectives.UnionWith(root.Usings);
+            foreach (var usingDirective in root.Usings)
+            {
+                if (combinedUsings.strings.Add(usingDirective.Name!.ToString()))
+                {
+                    combinedUsings.syntaxes.Add(usingDirective);
+                }
+            }
 
             foreach (var typeDecl in root.DescendantNodes().OfType<TypeDeclarationSyntax>())
             {
-                var fullName = NamespaceHelper.GetFullyQualifiedName(typeDecl);
-                combinedTypesDefined.Add(fullName);
+                if (!typeDecl.Modifiers.Any(SyntaxKind.PublicKeyword))
+                {
+                    continue;
+                }
+
+                var namespaceName = NamespaceHelper.GetNamespaceName(typeDecl);
```

Comment:
why did we previously need FQDN and now we just use namespace instead? Does that not break some other use cases?

Reply:
Type name is still used below. We're grouping namespaces and types in those namespaces.

Reply:
Alright just to clarify my question some more - FQDN would suggest like `RootNamespace.ChildNamespace` - so to me GetNamespaceFQDNName would return whole string where GetNamespaceName would return just ChildNamespace - from code it looks that's the case. I'm just wondering if that change doesn't break previous code that assumed it'll be FQND?

Reply:
Previously it was returning full namespace + type name, now it returns just full namespace.
I just tested and it works as I said.